symengine/serialize-cereal.h

```diff
@@ -0,0 +1,696 @@
+#pragma once
```

Comment:
This is non-standard. Use include guards instead?

Reply:
I'm curious: can you name a compiler which can compile symengine, but doesn't support `#pragma once`? 🙂

Reply:
I've seen the `#pragma once` used lately in C++. So far I always suggested to just use the old fashion include guards.

I am not against using `#pragma once`, it is definitely simpler. However I want to make sure we do not break the compilation.

@bjodah can you summarize the support in compilers for `#pragma once`? And why is it not standard?

Reply:
Good point. The only compiler I've seen is Oracle Development Studio which we have partial support for. https://github.com/symengine/symengine/issues/1646

Reply:
I don't know why it's not supported by the standard (the standard doesn't say much about the preprocessor from what I've seen). My question was genuine: I was curious about the existence of any compilers which lacks support for it (and I've seen multiple recommendations to prefer the pragma approach). But now we know: Oracle doesn't support it then. 🙂

Reply:
Do we want to support Oracle?

Reply:
It's not a big deal to keep using include guards instead of `pragma once`. I think we are using Oracle for R wrappers.

Reply:
Let's use the guards then.